src/services/TagBroadcastService.ts

```diff
@@ -0,0 +1,111 @@
+import { Commander } from "../TS3Connection";
+import { log } from "../Util";
+
+export class TagBroadcastService {
+    private readonly ZERO_WIDTH_SPACE = "\u200B";
+    private readonly COLOR_ACTIVE = Util.resolveColor("GREEN");
+    private readonly COLOR_INACTIVE = Util.resolveColor("RED");
+    private readonly COLOR_UNKNOWN = Util.resolveColor("GREY");
+
+    private client: BotgartClient;
+    private broadcastChannel: string;
+    private pingRole: string;
+
+    constructor(client: BotgartClient) {
+        this.client = client;
+        const config = getConfig().get();
+
+        this.broadcastChannel = config.ts_listener.broadcast_channel;
+        this.pingRole = config.ts_listener.ping_role;
+
+    }
+
+    async sendTagUpBroadcast(g: discord.Guild,
+                             commander: Commander,
+                             discordUser: discord.GuildMember | undefined,
+                             registration: undefined | Registration) {
+        // broadcast the message
+        const dchan: discord.TextChannel = <discord.TextChannel>g.channels.cache.find(c => c.name === this.broadcastChannel && c.type == "text");
+        if (!dchan) {
+            log("warning", `I was supposed to broadcast the commander message on guild '${g.name}' in channel '${this.broadcastChannel}', but no such channel was found there. Skipping.`);
+        } else {
+            const pingRole = g.roles.cache.find(r => r.name === this.pingRole);
+            const channelPath = commander.getTs3channelPath().map(value => `\`${value}\``).join(" ❯ ");
+            const name = discordUser?.displayName || commander.getTS3DisplayName();
+
+            const role = registration?.registration_role || "?";
+            const pingRoleMention = pingRole ? pingRole.toString() : "";
+
+            const embed = this.createEmbed(channelPath, commander);
+            const sendPromise = dchan.send(this.createMessage(name, role, pingRoleMention), embed);
+
+            if (discordUser?.user !== undefined) {
+                // user is known on discord -> is pingable
+                const userMention = discordUser!.user!.toString();
+                // replace message with version that links the user after it hast been sent initially.
+                // Editing does not trigger a notification.
+                return sendPromise.then(value => value.edit(this.createMessage(userMention, role, pingRoleMention), embed))
+            }
+            return sendPromise
+        }
+    }
+
+    private createEmbed(channelPath: string, commander: Commander, includeLink = true, color = this.COLOR_ACTIVE) {
+        const embed = new MessageEmbed();
+        let text = channelPath + " ❯ " + commander.getTS3DisplayName();
+        if (commander.getTs3joinUrl())
+            text += `\n\n [🔗 Join on TeamSpeak](${commander.getTs3joinUrl()} 'If you have TeamSpeak 3 installed, you can join by clicking here')`
```

Comment:
This might be a candidate for localisation.

Reply:
I don't really see how that would work: 2 Localizations on a single link ?

Reply:
I would probably add a field for each active localisation and follow up with the standalone URL, instead of working the URL into the English-only text.

Reply:
I tried showing the URL but it was not very visually appealing.
Adding more fields makes the message appear exorbitant large

Reply:
Could the more compact localisation variant that fits in a single line without flags which is used in achievement posts help in that case?
https://github.com/FlussuferOrga/Botgart/blob/master/src/commands/achievements/Achievements.ts#L159